artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java

```diff
@@ -772,6 +772,8 @@ public RoutingStatus route(final Message message,
 
       message.cleanupInternalProperties();
 
+      server.callBrokerPlugins(server.hasBrokerPlugins() ? plugin -> plugin.beforeMessageRoute(message, context, direct, rejectDuplicates) : null);
```

Comment:
in what situation this was not happening? paging?

Reply:
If there were messages sent to an address with no bindings the message would get dropped because of lack of bindings.  But with the old way the message route methods were only triggered if the routing status was OK.  With the change the methods are called even if there are no bindings but the status will say NO_BINDINGS, etc.